ngraph/test/preprocess.cpp

```diff
@@ -208,32 +221,87 @@ TEST(pre_post_process, test_2_inputs_basic) {
+    EXPECT_EQ(f->get_parameters().front()->get_output_tensor(0).get_names(), tensor_names);
+
+    auto result = std::make_shared<HostTensor>();
+    f->evaluate({result}, {make_host_tensor<ngraph::element::f32>(Shape{1, 2, 1, 3}, {5., 2., 7., 7., 8., -1.})});
+    auto result_val = read_vector<float>(result);
+    EXPECT_TRUE(all_close_f(std::vector<float>{2., 0., 1., 3., 2., -1.}, result_val));
+}
+
 TEST(pre_post_process, scale_vector_no_channels_layout) {
     auto f = create_simple_function(element::f32, Shape{1, 3, 224, 224});
     ASSERT_EQ(f->get_output_element_type(0), element::f32);
-    ASSERT_ANY_THROW(f = PrePostProcessor()
-                             .input(InputInfo()
-                                        .tensor(InputTensorInfo().set_layout("N?HW"))
-                                        .preprocess(PreProcessSteps().scale({0.1f, 0.2f, 0.3f})))
-                             .build(f));
+    ASSERT_THROW(f = PrePostProcessor()
+                         .input(InputInfo()
+                                    .tensor(InputTensorInfo().set_layout("N?HW"))
+                                    .preprocess(PreProcessSteps().scale({0.1f, 0.2f, 0.3f})))
+                         .build(f),
+                 ov::AssertFailure);
+}
+
+TEST(pre_post_process, scale_vector_channels_out_of_range) {
+    auto f = create_simple_function(element::f32, Shape{1, 3, 224, 224});
+    ASSERT_EQ(f->get_output_element_type(0), element::f32);
+    ASSERT_THROW(f = PrePostProcessor()
+                         .input(InputInfo()
+                                    .tensor(InputTensorInfo().set_layout("0123C"))
+                                    .preprocess(PreProcessSteps().scale({0.1f, 0.2f, 0.3f})))
+                         .build(f),
+                 ov::AssertFailure);
 }
 
 TEST(pre_post_process, mean_vector_no_layout) {
     auto f = create_simple_function(element::f32, PartialShape{Dimension::dynamic(), 3, 224, 224});
     ASSERT_EQ(f->get_output_element_type(0), element::f32);
-    ASSERT_ANY_THROW(
-        f = PrePostProcessor().input(InputInfo().preprocess(PreProcessSteps().mean({0.1f, 0.2f, 0.3f}))).build(f));
+    ASSERT_THROW(
+        f = PrePostProcessor().input(InputInfo().preprocess(PreProcessSteps().mean({0.1f, 0.2f, 0.3f}))).build(f),
+        ov::AssertFailure);
+}
+
+TEST(pre_post_process, mean_vector_dynamic_channels_shape) {
+    auto f = create_simple_function(
+        element::f32,
+        PartialShape{Dimension::dynamic(), Dimension::dynamic(), Dimension::dynamic(), Dimension::dynamic()});
+    ASSERT_EQ(f->get_output_element_type(0), element::f32);
+    ASSERT_THROW(f = PrePostProcessor()
+                         .input(InputInfo()
+                                    .tensor(InputTensorInfo().set_layout("NCHW"))
+                                    .preprocess(PreProcessSteps().mean({0.1f, 0.2f, 0.3f})))
+                         .build(f),
+                 ov::AssertFailure);
```

Comment:
@jane-intel is it valid case from `subtract` operation point of view? In a theory if we reshape such network to `[X 3 Y Z]` it's valid. While for `[X 4 Y Z]` we should throw an error on `reshape` stage

Reply:
Fixed, test is corrected. Now it allows such situation and will throw on evaluation stage if user's tensor mismatches with mean/scale